### DISCUSSION TITLE
Allow bitbucket app secret authentication with netlifycms

### DISCUSSION THREAD

**Discussion by ghost** [decaporg/decap-cms#6753] [Q&A]
Hi,

I would like to use Netlifycms for my documentation website (docusaurus) hosted on my company’s server. My company forces me to use bitbucket as code repo so I followed the steps here: https://www.netlifycms.org/docs/bitbucket-backend/

however, for security reason and because bitbucket workspace host many code repo across my organization, the security team does not allow me to create an OAuth consumer (managed at bitbucket workspace level) with read and write access but is suggesting to create an app secret (managed at bitbucket user level) instead

is it currently possible ? If not, any alternatives. 
thanks in advance for your help 
Thomas

**Comment by danurbanowicz** [comment]
There are two ways of authenticating with Bitbucket, explained in this comment on the Netlify Forums: https://answers.netlify.com/t/setting-up-config-yml-for-cms-connection-to-bitbucket/12464/6

Have you looked at either of those methods?

**Comment by ghost** [comment]
Thanks for the quick response. 

Correct me if I’m wrong but both ways described in the link above require me to create an oauth2 consumer in bitbucket, correct ? According to the page below, the scope of a oauth2 consumer is at workspace level and requires read and write access. https://support.atlassian.com/bitbucket-cloud/docs/use-oauth-on-bitbucket-cloud/

this is where I’m blocked because giving read-write access to the overall bitbucket workspace is not compliant with my company policies…

security team is asking for usage of app secrets/ password in bitbucket: https://support.atlassian.com/bitbucket-cloud/docs/app-passwords/

hope it helps
Thomas

**Comment by danurbanowicz** [comment]
Both documented methods of connecting to Bitbucket currently require oauth2 provider tokens, yes. There is no support for using Bitbucket app passwords that I am aware of.

Could you not create a new Bitbucket workspace for Decap CMS and then create an oauth2 consumer scoped to that workspace?

**Comment by ghost** [comment]
Unfortunately, I cannot do it as it’s not following standard company policy. I do not own the bitbucket app within my company and creating a new workspace just for 1 repo is not an option. Can you please add it as feature request ? 

If it’s not possible, then me and my team (10 to 15 users) will have to use another CMS, unfortunately…
Thanks

**Comment by martinjagodic** [comment]
I converted this into a discussion as it appears to be a question rather than an issue.
